From 775a2a8940c07f4af33a2a11bfa17e0257b427cb Mon Sep 17 00:00:00 2001
From: Florian Klink <flokli@flokli.de>
Date: Sat, 7 Mar 2020 22:40:27 +0100
Subject: [PATCH 16/19] kmod-static-nodes.service: Update ConditionFileNotEmpty

kmod loads modules from not only /lib/modules but also from
/run/booted-system/kernel-modules/lib/modules and
/run/current-system/kernel-modules/lib/module

Co-authored-by: Arian van Putten <arian.vanputten@gmail.com>
---
 units/kmod-static-nodes.service.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/units/kmod-static-nodes.service.in b/units/kmod-static-nodes.service.in
index 777e82d16b..9a5e05a1cc 100644
--- a/units/kmod-static-nodes.service.in
+++ b/units/kmod-static-nodes.service.in
@@ -12,7 +12,9 @@ Description=Create List of Static Device Nodes
 DefaultDependencies=no
 Before=sysinit.target systemd-tmpfiles-setup-dev.service
 ConditionCapability=CAP_SYS_MODULE
-ConditionFileNotEmpty=/lib/modules/%v/modules.devname
+ConditionFileNotEmpty=|/lib/modules/%v/modules.devname
+ConditionFileNotEmpty=|/run/booted-system/kernel-modules/lib/modules/%v/modules.devname
+ConditionFileNotEmpty=|/run/current-system/kernel-modules/lib/modules/%v/modules.devname
 
 [Service]
 Type=oneshot
-- 
2.33.1

